EESSI-install-software.sh

```diff
@@ -147,6 +147,24 @@ else
   mkdir -p ${EESSI_PREFIX}/software/${EESSI_OS_TYPE}/${EESSI_SOFTWARE_SUBDIR_OVERRIDE}
 fi
 
+# if we run the script for the first time, e.g., to start building for a new
+#   stack, we need to ensure certain files are present in
+#   ${EESSI_PREFIX}/software/${EESSI_OS_TYPE}/${EESSI_SOFTWARE_SUBDIR_OVERRIDE}
+# - .lmod/lmodrc.lua
+# - .lmod/SitePackage.lua
```

Comment:
@trz42 Why do these need to be in place exactly?

Reply:
It seems these files are needed when `init/eessi_environment_variables` is sourced. The actual build may still work, but the issue is that the non-existence of these files creates `ERROR` messages in the slurm output file. The `bot/check-build.sh` thus considers the job being failed.

Reply:
OK, also have to do this here probably means that the same calls at the end of the script are just doing it too late, and it should be moved here (rather than copied)?

I'm happy to merge this as if for now, there's no negative impact when having it on both places, but it's silly, and duplicated code, so we should clean that up afterwards.

Reply:
I think we need to have the files created early. As it is now it creates them at the beginning if they don't exist. At the end they are created if they don't exist or if they have changed. So if we check for the change at the beginning we may remove the creation at the end.